### PR TITLE
Issue #77 - Externalize the target platform into definition file

### PR DESCRIPTION
**Pull request by abstratt**
- Externalized target platform using a target definition file for easier dev set-up - see https://wiki.eclipse.org/PDE/Target_Definitions

### CODE REVIEW

File: org.modeldriven.alf.eclipse.moka/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -17,7 +17,7 @@ Require-Bundle: org.eclipse.uml2;bundle-version="4.0.2";resolution:=optional,
  org.eclipse.papyrus.moka;bundle-version="2.0.100",
  org.eclipse.papyrus.moka.fuml,
  org.eclipse.papyrus.moka.fuml.interfaces;bundle-version="2.0.0",
- org.eclipse.papyrus.moka.fuml.standardlibrary;bundle-version="2.0.0",
+ org.eclipse.papyrus.moka.fuml.standardlibrary;bundle-version="2.0.100",
```
Comment (abstratt) [inline org.modeldriven.alf.eclipse.moka/META-INF/MANIFEST.MF]: This was necessary as there was an API feature added after 2.0.0 that we rely on (Div_ class).

**Comment by abstratt** [comment]
Thanks @seidewitz, that was something I was unsure about. Done.

**Comment by abstratt** [comment]
@seidewitz I tried to infer the versions of upstream projects based on the minimum bundle versions you had specified. Is this giving you different results than the approach you used so far (resulting in unexpected - too old or to new - versions of dependencies)? 

Updating this to a more recent set of dependencies makes sense. Maybe we could that later as a separate task, assuming the dependencies proposed here are not incorrect?

**Comment by seidewitz** [comment]
Actually, you did pretty much get the versions right, and I am, in fact, still using Oxygen for RI development. You can probably also add http://eclipse-javacc.sourceforge.net (SF JavaCC Eclipse plugin feature 1.5.33) to the platform.

I originally stayed on Oxygen/4.7 (and, in fact, until recently, was still testing on 4.6, too) for compatibility with a user that is no longer relevant. I have been thinking for a while that it is finally time to move on to the current Eclipse version, but haven't had a chance to do so. However, doing that as a separate task probably makes sense -- it is lower priority than making progress on Issue #75.

**Comment by abstratt** [comment]
@seidewitz The target platform needs to contain the bare minimum bundles that the project's own bundles are compiled against. it does not affect the functionality available to the IDE during development time. For instance, we can still use an old version of Eclipse as the target platform (because the Alf RI does not require a newer version) and use the latest Eclipse as our development environment.

Re: the JavaCC plug-in for Eclipse - since JavaCC itself or the JavaCC Eclipse plug-in are not used for compiling the code, at this time it seems it is not needed in the target platform. Do you see a reason for us to include it?

Re: versions - the target platform should reflect our minimum requirements, so making it rely on recent versions of Eclipse, Eclipse UML2 or Papyrus mean dropping compatibility for earlier versions, even if as recent as last year's. No problem with that, if that is really what  you are looking for. At least for the Eclipse Platform, I would try to have the oldest possible version the Alf RI can work on as a prerequisite.

**Comment by seidewitz** [comment]
Not sure if including the Javacc plugin in the platform might be necessary if we better automate the build process in the future. But that's not important for now.

**Pull request merged**